main.cpp

```diff
@@ -115,17 +147,25 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T tInst1( 1, "T struct instance 1" );               //6
+    T tInst2( 2, "T struct instance 1" );               //6
+    
+    CheckPointer f;                                           //7
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    auto* smaller = f.compare( &tInst1, &tInst2);                    if(smaller == nullptr) //8
+    {
+        std::cout << "The pointer in null! " << std::endl;
+    }
+    else
+    {
+        std::cout << "the smaller one is << " << smaller->name << std::endl;
+    }
```

Comment:
Don't write `if(smaller == nullptr)` on the same line as another expression
Also:
I recently updated the instructions for this line in the project: 
```
 9) correct the cout that uses smaller's member variable with some safe pointer usage.
    the possible return values of the compare() function should give you a hint.
    Also: if smaller == nullptr, make the cout statement explain the reasons that `f.compare()` 
    might return nullptr.  
    hint: There are multiple reasons
```
- [ ] revise